tests/test_generator.py

```diff
@@ -0,0 +1,75 @@
+from deepinterpolation.generic import JsonSaver, ClassLoader
+import os
+import pathlib
+
+
+def test_generator_tif_creation(tmp_path):
+
+    generator_param = {}
+
+    # We are reusing the data generator for training here.
+    generator_param["type"] = "generator"
+    generator_param["name"] = "SingleTifGenerator"
+    generator_param["pre_post_frame"] = 30
+    generator_param["pre_post_omission"] = 0
+    generator_param[
+        "steps_per_epoch"
+    ] = -1
+
+    generator_param["train_path"] = os.path.join(
```

Comment:
just FYI, pathlib lets you get around using `os` almost all of the time.
For example, here, you could do:
```
generator_param["train_path"] =  str(pathlib.Path(__file__).parent.parent /
                                                          "sample_data" / "ophys_tiny_761605196.tif")
```
there's a good table for translating between the 2:
https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module